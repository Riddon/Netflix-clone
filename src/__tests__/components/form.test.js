import React from 'react';
import { render } from '@testing-library/react';
import { Form } from "../../components/";

jest.mock('react-router-dom');

describe('<Form />', () => {
    it('renders the <Form/> with populated data',  () => {

        const { container, getByText, getByPlaceholderText } = render(
            <Form>
                <Form.Title>Sign In Now.</Form.Title>

                <Form.Base>
                    <Form.Input placeholder="Email address" onChange={() => {}}/>

                    <Form.Input type="password" placeholder="Password" onChange={() => {}}/>

                    <Form.Submit disabled type="submit">
                        Sign In
                    </Form.Submit>
                </Form.Base>

                <Form.Text>
                    New to Netflix?
                    <Form.Link to='/signup'>Sign up now.</Form.Link>
                </Form.Text>
                <Form.TextSmall>
                    This page is protected by Google reCAPTCHA to ensure you`re not a bot. Lear more.
                </Form.TextSmall>
            </Form>
        );

        expect(getByText('Sign In Now.')).toBeTruthy();
        expect(getByText('Sign In')).toBeTruthy();
        expect(getByText('Sign In').disabled).toBeTruthy();
        expect(getByText('This page is protected by Google reCAPTCHA to ensure you`re not a bot. Lear more.')).toBeTruthy();

        expect(getByPlaceholderText('Email address')).toBeTruthy();
        expect(getByPlaceholderText('Password')).toBeTruthy();

        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders the <Form /> with an error', () => {
        const error = 'Your email address is already being used';

        const { container, getByText, queryByText } = render(
            <Form>
                {error && <Form.Error>{error}</Form.Error>}
                <Form.Submit type="submit">Sign In</Form.Submit>
            </Form>
        );

        expect(getByText('Your email address is already being used')).toBeTruthy();
        expect(queryByText('Sign In').disabled).toBeFalsy();
        expect(container.firstChild).toMatchSnapshot();
    });

});