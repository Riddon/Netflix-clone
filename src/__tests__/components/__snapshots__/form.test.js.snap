// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> renders the <Form /> with an error 1`] = `
<div
  class="sc-bdvvaa djRlgr"
>
  <div
    class="sc-dkPtyc kxAWBk"
  >
    Your email address is already being used
  </div>
  <button
    class="sc-furvIG ekqsPW"
    type="submit"
  >
    Sign In
  </button>
</div>
`;

exports[`<Form /> renders the <Form/> with populated data 1`] = `
<div
  class="sc-bdvvaa djRlgr"
>
  <h1
    class="sc-hKwCoD bikTfr"
  >
    Sign In Now.
  </h1>
  <form
    class="sc-gsDJrp gTRyZq"
  >
    <input
      class="sc-iCfLBT lbiCQg"
      placeholder="Email address"
    />
    <input
      class="sc-iCfLBT lbiCQg"
      placeholder="Password"
      type="password"
    />
    <button
      class="sc-furvIG ekqsPW"
      disabled=""
      type="submit"
    >
      Sign In
    </button>
  </form>
  <p
    class="sc-eCImvq fyyRrz"
  >
    New to Netflix?
    <a
      class="sc-gKckTs iuEBPy"
      href="/signup"
    >
      Sign up now.
    </a>
  </p>
  <p
    class="sc-jRQAMF gidvgx"
  >
    This page is protected by Google reCAPTCHA to ensure you\`re not a bot. Lear more.
  </p>
</div>
`;
